src/copy.rs

```diff
@@ -45,6 +47,21 @@ where
     }
 }
 
+// TcpStreamSplitter is a specialized BufferedSplitter for TcpStream, which is more efficient than the generic
+// `tokio::io::split`. The generic method involves locking to access the read and write halves
```

Comment:
If I'm reading the upstream code correctly, we are saving the mutex but losing (a bit) on the heap allocation side?

Since we have to  use `into_split` which the generic `tokio:io:split` func avoids.

Reply:
Choices are:

* tcp_stream.split: no lock,no heap
* tcp-stream.into_split: no lock, heap
* tokio::io::split: lock, heap

Reply:
Right, and former produces splits that can't be dropped into a spawn. Fair enough.